### ISSUE TITLE
Android 4.2.1 specific crash

### ISSUE THREAD

**Issue by Rolf-Smit** [bitstadium/HockeySDK-Android#269]
In our app we have a crash, that occurs within the HockeyApp SDK, ironically this crash is logged by HockeyApp so that's nice.

This crash only occurs on Android 4.2.1 devices and more specifically Samsung devices, quite some users are affected by this crash. I did some research and it seems like this crash is caused by a bug in the  http framework used by Android, but even though this is the case it is still possible to fix this on the SDK side.

```
Package: **********
Version Code: **********
Version Name: **********
Android: 4.1.2
Android Build: JZO54K.I8260XXAPA1
Manufacturer: samsung
Model: GT-I8260
Thread: AsyncTask #3-914
CrashReporter Key: **********
Start Date: **********
Date: **********

java.lang.RuntimeException: An error occured while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:299)
	at java.util.concurrent.FutureTask$Sync.innerSetException(FutureTask.java:273)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:124)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:307)
	at java.util.concurrent.FutureTask.run(FutureTask.java:137)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1076)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:569)
	at java.lang.Thread.run(Thread.java:856)
Caused by: java.lang.NullPointerException
	at org.apache.harmony.xnet.provider.jsse.OpenSSLSocketFactoryImpl.createSocket(OpenSSLSocketFactoryImpl.java:94)
	at libcore.net.http.HttpConnection.setupSecureSocket(HttpConnection.java:197)
	at libcore.net.http.HttpsURLConnectionImpl$HttpsEngine.makeSslConnection(HttpsURLConnectionImpl.java:478)
	at libcore.net.http.HttpsURLConnectionImpl$HttpsEngine.connect(HttpsURLConnectionImpl.java:433)
	at libcore.net.http.HttpEngine.sendSocketRequest(HttpEngine.java:289)
	at libcore.net.http.HttpEngine.sendRequest(HttpEngine.java:239)
	at libcore.net.http.HttpURLConnectionImpl.connect(HttpURLConnectionImpl.java:80)
	at libcore.net.http.HttpURLConnectionImpl.getOutputStream(HttpURLConnectionImpl.java:188)
	at libcore.net.http.HttpsURLConnectionImpl.getOutputStream(HttpsURLConnectionImpl.java:281)
	at net.hockeyapp.android.metrics.Sender.getWriter(Sender.java:408)
	at net.hockeyapp.android.metrics.Sender.logRequest(Sender.java:328)
	at net.hockeyapp.android.metrics.Sender.send(Sender.java:144)
	at net.hockeyapp.android.metrics.Sender.sendAvailableFiles(Sender.java:126)
	at net.hockeyapp.android.metrics.Sender$1.doInBackground(Sender.java:86)
	at net.hockeyapp.android.metrics.Sender$1.doInBackground(Sender.java:82)
	at android.os.AsyncTask$2.call(AsyncTask.java:287)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:305)
	... 4 more
```

**Comment by TroubleMakerBen** [comment]
hey Rolf,

thanks for getting in touch and reporting this. Which version of the HockeySDK are you using?

Best,
Benny

**Comment by Rolf-Smit** [comment]
Sorry for the late response, I did not get a notification in my mail box.

`net.hockeyapp.android:HockeySDK:4.1.4`

**Comment by TroubleMakerBen** [comment]
We just released HockeySDK 5.0.0 which fixes the issue, so I'm closing this.
